tests/test_utils/test_mpi_utils_2d.py

```diff
@@ -46,134 +46,53 @@ def test_mpi_field_io_gather_scatter(n_values, precision):
         )
 
 
-@pytest.mark.mpi(group="MPI_utils")
+@pytest.mark.mpi(group="MPI_utils", min_size=2)
+@pytest.mark.parametrize("ghost_size", [1, 2, 3])
 @pytest.mark.parametrize("precision", ["single", "double"])
-@pytest.mark.parametrize("n_values", [16])
-def test_mpi_ghost_blocking_communication(n_values, precision):
+@pytest.mark.parametrize("rank_distribution", [(1, 0), (0, 1)])
+@pytest.mark.parametrize("aspect_ratio", [(1, 1), (1, 2), (2, 1), (1, 1)])
```

Comment:
this is a good idea, maybe we should add this to the diffusion kernel as well?

Reply:
Yeah that's a good idea. Implemented in [a0bce0a](https://github.com/fankiat/sopht-mpi/pull/5/commits/a0bce0a66151148448868c60bd9cc10fbf15d5b8)
Also good catch on the redundant aspect ratio. Thanks!